slsa_for_models/gcp/README.md

```diff
@@ -0,0 +1,247 @@
+      ```    
+
+7. Generate an encrypted x509 keypair and save it as a Kubernetes secret:
+   
+   ```shell
+   cosign generate-key-pair k8s://tekton-chains/signing-secrets
+   ```
+ 
+8. Apply the `Tasks`:
+
+   ```shell
+   kubectl apply -f slsa_for_models/gcp/tasks
+   ```
+
+9. Apply the `Pipeline`:
+
+   ```shell
+   kubectl apply -f slsa_for_models/gcp/pipeline.yml
+   ```
+ 
+10. Create a generic repository in Artifact Registry:
+
+    1. Set the `REPOSITORY_NAME` environment variable to a name of your choice:
+
+       ```shell
+       export REPOSITORY_NAME=ml-artifacts
+       ```
+
+    2. Set the `LOCATION` environment variable to a [location] of your choice:
+
+       ```shell
+       export LOCATION=us
+       ```
+       
+    3. Create a generic repository:
+        ```shell
+        gcloud artifacts repositories create $REPOSITORY_NAME \
+         --location=$LOCATION \
+         --repository-format=generic
+        ```   
+   
+    4. If you set a different repository name and location from the example above, make sure to modify the `Parameter`
+    named 'model-storage' in the `PipelineRun` with your own values. 
+
+11. Execute the `PipelineRun`:
+
+    ```shell
+    kubectl create -f slsa_for_models/gcp/pipelinerun.yml
+    ```
+
+12. Observe the `PipelineRun` execution:
+
+    ```shell
+    export PIPELINERUN_NAME=$(tkn pr describe --last --output jsonpath='{.metadata.name}')
+    tkn pipelinerun logs $PIPELINERUN_NAME --follow
+    ``` 
+
+13. When the `PipelineRun` succeeds, view its status:
+
+    ```shell
+    kubectl get pipelinerun $PIPELINERUN_NAME --output yaml
+    ```
+
+14. View the transparency log entry in the public [Rekor][rekor] instance:
+
+   ```shell
+   export TLOG_ENTRY=$(tkn pr describe $PIPELINERUN_NAME --output jsonpath="{.metadata.annotations.chains\.tekton\.dev/transparency}")
+   open $TLOG_ENTRY
+   ```
+   
+15. Retrieve the attestation from the `PipelineRun` which is stored as a base64-encoded annotation:
+
+   ```shell
+   export PIPELINERUN_UID=$(tkn pr describe $PIPELINERUN_NAME --output  jsonpath='{.metadata.uid}')
+   tkn pr describe $PIPELINERUN_NAME --output jsonpath="{.metadata.annotations.chains\.tekton\.dev/signature-pipelinerun-$PIPELINERUN_UID}" | base64 -d > pytorch_model.pth.intoto.jsonl 
+   ```
+
+16. View the attestation:
+
+   ```shell
+   cat pytorch_model.pth.intoto.jsonl | tr -d '\n' | pbcopy
+   pbpaste | jq '.payload | @base64d | fromjson'
+   ```
+17. Download the model:
+
+   ```shell
+   export MODEL_VERSION=$(tkn pr describe slsa-for-models-pr --output jsonpath='{.status.results[1].value.digest}' | cut -d ':' -f 2)
+   gcloud artifacts generic download \
+   --package=pytorch-model \
+   --repository=$REPOSITORY_NAME \
+   --destination=. \
+   --version="${MODEL_VERSION:7}"
+   ```
+
+18. Verify the attestation:
+
+   ```shell
+   cosign verify-blob-attestation \
+    --key k8s://tekton-chains/signing-secrets \
+    --signature pytorch_model.pth.intoto.jsonl \
```

Comment:
nit: shall we update to `pytorch_model.pth.build-slsa` instead? This makes the extension clearer and does not expose the underlying implementation (intoto, json, etc). If we ever change the underlying format, we don't break users because the underlying format is _not_ exposed.

/cc @mihaimaruseac

Reply:
Done! 

I'd named it to follow the GHA example, I think that should be updated as well: https://github.com/google/model-transparency/blob/cf1c033f6033ebf8df988539d1050a0a5c8fa928/slsa_for_models/README.md?plain=1#L62

Reply:
+1